### PR TITLE
move hardcoded argo tunnel url to env var

### PR DESCRIPTION
**Pull request by bellaweo**
remove hardcoded kitt.run domain from tunnel url, allow user to set dynamically in .env file
add traefik config settings to enable letsencrypt ssl cert generation via dns challenge

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -22,6 +22,10 @@ services:
       - --certificatesResolvers.le.acme.dnsChallenge.provider=cloudflare
       - --certificatesResolvers.le.acme.email=${CF_API_EMAIL}
       - --certificatesResolvers.le.acme.storage=/etc/traefik/acme/acme.json
+      - --certificatesResolvers.le.acme.dnschallenge=true
+      - --certificatesResolvers.le.acme.dnschallenge.provider=cloudflare
```
Comment (jojomomojo) [inline docker-compose.yml:26]: dupe, this is set in line 22

File: docker-compose.yml [hunk 2]
```diff
@@ -73,7 +77,7 @@ services:
     volumes:
       - ${HOME}/.cloudflared:/etc/cloudflared
     environment:
-      - TUNNEL_URL=https://traefik.kitt.run
+      - TUNNEL_URL=https://${KITT_ARGO_URL}
       - TUNNEL_HOSTNAME=${KITT_ARGO_TUNNEL}
```
Comment (jojomomojo) [inline docker-compose.yml:81]: Can we use KITT_TUNNEL_URL and KITT_TUNNEL_HOSTNAME as the env var names?

Comment (bellaweo) [reply docker-compose.yml:81]: I think TUNNEL_URL and TUNNEL_HOSTNAME is hardcoded in cloudflared
https://developers.cloudflare.com/argo-tunnel/reference/arguments/#url

Comment (bellaweo) [reply docker-compose.yml:80]: oh i see you want to remove ARGO from the env vars

Comment (jojomomojo) [reply docker-compose.yml:80]: I just made them up for no good reason... should have used the CF env vars as a hint.

**Pull request merged**